03_cross_cancer_classification/run_cross_cancer_flip_labels.py

```diff
@@ -0,0 +1,239 @@
+    train_cross_cancer,
+    evaluate_cross_cancer
+)
+import pancancer_evaluation.utilities.data_utilities as du
+import pancancer_evaluation.utilities.file_utilities as fu
+
+def process_args():
+    p = argparse.ArgumentParser()
+    p.add_argument('--debug', action='store_true',
+                   help='use subset of data for fast debugging')
+    p.add_argument('--holdout_percent_vals', nargs='+', default=[0.2], type=float,
+                   help='percent of labels to hold out of test set, '
+                        'float between 0 and 1')
+    p.add_argument('--log_file', default=None,
+                   help='name of file to log skipped cancer types to')
+    p.add_argument('--results_dir', default=cfg.results_dir,
+                   help='where to write results to')
+    p.add_argument('--seed', type=int, default=cfg.default_seed)
+    p.add_argument('--subset_mad_genes', type=int, default=cfg.num_features_raw,
+                   help='if included, subset gene features to this number of '
+                        'features having highest mean absolute deviation')
+    p.add_argument('--verbose', action='store_true')
+    args = p.parse_args()
+
+    args.results_dir = Path(args.results_dir).resolve()
+
+    if args.log_file is None:
+        args.log_file = Path(args.results_dir, 'log_skipped_cc.tsv').resolve()
+
+    return args
+
+if __name__ == '__main__':
+
+    # process command line arguments
+    args = process_args()
+
+    # create results dir if it doesn't exist
+    args.results_dir.mkdir(parents=True, exist_ok=True)
+
+    # create empty log file if it doesn't exist
+    log_columns = [
+        'train_identifier',
+        'percent_holdout',
+        'shuffle_labels',
+        'skip_reason'
+    ]
+    if args.log_file.exists() and args.log_file.is_file():
+        log_df = pd.read_csv(args.log_file, sep='\t')
+    else:
+        log_df = pd.DataFrame(columns=log_columns)
+        log_df.to_csv(args.log_file, sep='\t')
+
+    tcga_data = TCGADataModel(seed=args.seed,
+                              subset_mad_genes=args.subset_mad_genes,
+                              verbose=args.verbose,
+                              debug=args.debug)
+
+    # use 5 easily predictable driver genes
+    genes_df = du.load_vogelstein()
+    sampled_genes = ['KRAS', 'TP53', 'BRAF', 'EGFR', 'PTEN']
+
+    # and use all cancer types in TCGA
+    sample_info_df = du.load_sample_info(args.verbose)
+    tcga_cancer_types = list(np.unique(sample_info_df.cancer_type))
+
+    # identifiers have the format {gene}_{cancer_type}
+    identifiers = ['_'.join(t) for t in it.product(sampled_genes,
+                                                   tcga_cancer_types)]
+
+    # create output directory
+    output_dir = Path(args.results_dir, 'cross_cancer_flip_labels').resolve()
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    for shuffle_labels in (False, True):
+
+        print('shuffle_labels: {}'.format(shuffle_labels))
```

Comment:
I'm not sure if this is a debug print statement or part of the desired output. If it's for debugging you can probably remove it

Reply:
It's part of the desired output - when my experiments are running it prints above the progress bars from `tqdm` so I can tell whether I'm using shuffled or original labels.

I guess I could make it into another (nested) progress bar, but I already have two here, and it seems to me like having a progress bar for only two steps wouldn't provide much information.

Reply:
That makes sense! Was just making sure it was intended